src/auto_archiver/modules/opentimestamps_enricher/opentimestamps_enricher.py

```diff
@@ -0,0 +1,184 @@
+import os
+import datetime
+
+from loguru import logger
+import opentimestamps
+from opentimestamps.calendar import RemoteCalendar, DEFAULT_CALENDAR_WHITELIST
+from opentimestamps.core.timestamp import Timestamp, DetachedTimestampFile
+from opentimestamps.core.notary import PendingAttestation, BitcoinBlockHeaderAttestation
+from opentimestamps.core.op import OpSHA256
+from opentimestamps.core import serialize
+from auto_archiver.core import Enricher
+from auto_archiver.core import Metadata, Media
+from auto_archiver.utils.misc import calculate_file_hash
+
+class OpentimestampsEnricher(Enricher):
+
+    def enrich(self, to_enrich: Metadata) -> None:
+        url = to_enrich.get_url()
+        logger.debug(f"OpenTimestamps timestamping files for {url=}")
+
+        # Get the media files to timestamp
+        media_files = [m for m in to_enrich.media if m.filename and not m.get("opentimestamps")]
+        if not media_files:
+            logger.warning(f"No files found to timestamp in {url=}")
+            return
+
+        timestamp_files = []
+        for media in media_files:
+            try:
+                # Get the file path from the media
+                file_path = media.filename
+                if not os.path.exists(file_path):
+                    logger.warning(f"File not found: {file_path}")
+                    continue
+                
+                # Create timestamp for the file - hash is SHA256
+                # Note: ONLY SHA256 is used/supported here. Opentimestamps supports other hashes, but not SHA3-512
+                # see opentimestamps.core.op
```

Comment:
FYI, for the purpose of time-stamping it'd be truly remarkable if SHA256 was ever not good enough. Even SHA1 is still secure for use as a hash function when timestamping: https://petertodd.org/2017/sha1-and-opentimestamps-proofs